src/features/security-cloud-game/__tests__/DragCloud.test.js

```diff
@@ -1,20 +1,114 @@
 import React from 'react';
+import Draggable from 'react-draggable';
 
-import DragCloud from '../DragCloud';
+import { DragCloudBase as DragCloud } from '../DragCloud';
+
+const intl = {
+    formatMessage: message => message.defaultMessage,
+};
 
 describe('features/security-cloud-game/DragCloud', () => {
     test('should correctly render', () => {
-        const component = shallow(<DragCloud cloudSize={100} position={{ x: 10, y: 20 }} />);
-
-        expect(component.hasClass('drag-cloud')).toBe(true);
-        expect(component.prop('style')).toEqual({
-            top: '20px',
-            left: '10px',
-        });
-        expect(component.find('IconCloud').length).toEqual(1);
-        expect(component.find('IconCloud').prop('height')).toEqual(100);
-        expect(component.find('IconCloud').prop('width')).toEqual(100);
-        expect(component.find('IconCloud').prop('filter')).toBeDefined();
-        expect(component.find('IconCloud').prop('filter').id).toEqual('drop-shadow');
+        const wrapper = shallow(
+            <DragCloud
+                gameBoardSize={{ height: 500, width: 500 }}
+                cloudSize={100}
+                gridTrackSize={20}
+                intl={intl}
+                position={{ x: 10, y: 20 }}
+            />,
+        );
+
+        const draggable = wrapper.find(Draggable);
+        expect(draggable.prop('position')).toEqual({ x: 10, y: 20 });
+        expect(draggable.find('.bdl-DragCloud').length).toEqual(1);
+
+        const iconCloud = wrapper.find('IconCloud');
+        expect(iconCloud.prop('height')).toEqual(100);
+        expect(iconCloud.prop('width')).toEqual(100);
+        expect(iconCloud.prop('title')).toEqual('Cloud object');
+        expect(iconCloud.prop('filter')).toBeDefined();
+        expect(iconCloud.prop('filter').id).toEqual('drop-shadow');
+    });
+
+    test('should handle keyboard navigation correctly', () => {
+        const updatePosition = jest.fn();
+        const onDrop = jest.fn();
+        const updateLiveText = jest.fn();
+
+        const wrapper = mount(
+            <DragCloud
+                gameBoardSize={{ height: 500, width: 500 }}
+                cloudSize={50}
+                gridTrackSize={10}
+                intl={intl}
+                onDrop={onDrop}
+                position={{ x: 20, y: 10 }}
+                updateLiveText={updateLiveText}
+                updatePosition={updatePosition}
+            />,
+        );
+
+        // grab
+        wrapper.find('.bdl-DragCloud').simulate('keydown', { key: ' ' });
+        expect(wrapper.find('.bdl-DragCloud').hasClass('is-moving')).toBe(true);
+        expect(updateLiveText).toHaveBeenLastCalledWith(
```

Comment:
could you change `toHaveBeenLastCalledWith` to `lastCalledWith`? there's a preference in this repo to use the aliases of the `toHaveBeen[...]` methods.

e.g. `toBeCalled` instead of `toHaveBeenCalled`

Reply:
Got it. Good to know! I changed all to use the aliases.